### DISCUSSION TITLE
Missing `handle` argument on `MenuItem`

### DISCUSSION THREAD

**Discussion by suiramdev** [Shopify/storefront-api-feedback#129] [Ideas & Requests]
While developing with Hydrogen my navigation bar, I realized that in a `MenuItem` element, there is only an `url` redirecting.
It is annoying for the Hydrogen based websites that it is given an URL with our Shopify's website. We are not always using Shopify as a host.

I had, to counter it, get the `handle` myself like the following:
```js
{items.map((item) => {
  const { pathname } = new URL(item.url || ""); // <------

  return (
    <Link
      to={`${pathname}`}
      className={clsx(
        "flex justify-center items-center px-10 py-5 bg-default-300 text-center hover:text-primary hover:no-underline",
        pathname.split("/").pop() === (handle || "") && "font-bold text-primary"
      )}
      key={item.id}
    >
      {item.title}
    </Link>
  );
})}
```
In my opinion, `MenuItem` should have a `handle` argument.

**Comment by vixdug** [comment]
Hey @suiramdev 

Can you give more information on what you are looking to accomplish? Where are your menu items pointing to? The parent menu has a handle in the admin. The `menuItems` however are usually identified by name / URL. Are you looking for something like: 

```
				{
					"id": "gid:\/\/shopify\/MenuItem\/12345678",
					"url": "https:\/\/1234.myshopify.com\/collections\/all"
                                         "handle": "/collections/all" 
				}

**Reply by suiramdev** [reply]
I am exactly looking for something like this.
From what I've seen, there is no `path name` given to the menu items. Only a URL, pointing to the Shopify-hosted website route (e.g: https://1234.myshopify.com/collections/all)
On the `menuItems`, it could have:
```json
{
	"id": "gid:\/\/shopify\/MenuItem\/12345678",
	"url": "https:\/\/1234.myshopify.com\/collections\/all"
  	"handle": "all", <--
	"pathname": "/collections/all" <--
}
```
So that, on Hydrogen websites, we don't need to split the `URL` element ourselves to only get the pathname/handle for our `<Link />` components, or whatever.

I don't know how to make it clearer.

You have the example comparing the `Collection` property and the `menuItem` :
The `collection` has the `handle` property, while the `menuItem` has no identifier for the page name.

**Reply by vixdug** [reply]
Understood, I'll bring the feedback up with the team. Thank you!
